src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

```diff
@@ -940,13 +1009,24 @@ function discoverController(
     $scope.minimumVisibleRows = $scope.hits;
   };
 
-  $scope.updateSavedQueryId = newSavedQueryId => {
-    if (newSavedQueryId) {
-      $state.savedQuery = newSavedQueryId;
-    } else {
-      delete $state.savedQuery;
-    }
-    $state.save();
+  $scope.onSavedQuerySaved = savedQuery => {
+    $scope.savedQuery = savedQuery;
+    updateStateFromSavedQuery(savedQuery);
+  };
+
+  $scope.onSavedQueryUpdated = savedQuery => {
+    $scope.savedQuery = { ...savedQuery };
+    updateStateFromSavedQuery(savedQuery);
+  };
+
+  $scope.onSavedQueryCleared = () => {
+    //reset filters and query string, remove savedQuery from state
+    const state = {
+      ...appStateContainer.getState(),
+      ...{ query: { ...$scope.state.query, ...{ query: '' } }, filters: [] },
+    };
```

Comment:
IMHO, this could look like:

```
const state = {
      ...appStateContainer.getState(),
      query: defaultQuery,
      filters: [],
    };
```

where the `defaultQuery` is initialized in upper scope:

![image](https://user-images.githubusercontent.com/31325372/75241295-e5e79a00-57d6-11ea-9bfe-b230e22dbd2b.png)

More than that, I would incapsulate it inside the `appStateContainer` transition function, like `appStateContainer.transitions.removeSavedQuery(defaultQuery)`.

You could see the example in visualize. More than that, the behavior in visualize is a bit different: on removal of saved query global filters are stayed in search.

Reply:
I've introduced a `getDefaultQuery` function for now, which improves that, else I don't find time to review a pretty visualize PR in my pipeline :)